File: trunk/openrest4j-api/src/com/googlecode/openrest/Payment.java
package com.googlecode.openrest;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Payment implements Serializable {
    /** Cash payment. */
    public static final String PAYMENT_TYPE_CASH = "cash";
    /** Credit card payment. */
    public static final String PAYMENT_TYPE_CREDIT = "credit";
    /** Payment by 10bis card (@see www.10bis.co.il). */
    public static final String PAYMENT_TYPE_10BIS = "10bis";

    /** All payment methods, in ascending alphabetic order! */
    public static final List<String> ALL_PAYMENT_TYPES = Arrays.asList(new String[] {
        PAYMENT_TYPE_10BIS, PAYMENT_TYPE_CASH, PAYMENT_TYPE_CREDIT
    });

    public Payment(String type, Integer amount, CreditCard card) {
        this.type = type;
        this.amount = amount;
        this.card = card;
    }

    /** Default constructor for JSON deserialization. */
    public Payment() {}

    /** Payment type. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String type;

    /** Amount to pay. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Integer amount = 0;

    /** Credit card details (not valid for PAYMENT_TYPE_CASH) */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public CreditCard card;

    private static final long serialVersionUID = 1L;
}

File: openrest4net/openrest4net-api/com/googlecode/openrest/Payment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace com.googlecode.openrest
{
    public class Payment
    {
        /** Cash payment. */
        public const string PAYMENT_TYPE_CASH = "cash";
        /** Credit card payment. */
        public const string PAYMENT_TYPE_CREDIT = "credit";
        /** Payment by 10bis card (@see www.10bis.co.il). */
        public const string PAYMENT_TYPE_10BIS = "10bis";

        /** All payment methods, in ascending alphabetic order! */
        public static readonly IList<string> ALL_PAYMENT_TYPES =
            new List<string> { PAYMENT_TYPE_10BIS, PAYMENT_TYPE_CASH, PAYMENT_TYPE_CREDIT };

        public Payment(string type, int amount, CreditCard card)
        {
            this.type = type;
            this.amount = amount;
            this.card = card;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        public Payment() { }

        /** Payment type. */
        public string type;

        /** Amount to pay. */
        public int? amount = 0;

        /** Credit card details (valid only if type is PAYMENT_TYPE_CREDIT) */
        public CreditCard card;
    }
}
